app/src/main/java/com/infomaniak/drive/utils/Utils.kt

```diff
@@ -235,22 +235,57 @@ object Utils {
         }
     }
 
+    fun Context.openWith(uri: Uri, type: String?, flags: Int) {
+        startActivityFor(openWithIntent(uri, type, flags))
+    }
+
     fun Context.openWith(file: File, userDrive: UserDrive = UserDrive()) {
+        startActivityFor(openWithIntent(file, userDrive))
+    }
+
+    private fun Context.startActivityFor(openWithIntent: Intent) {
         try {
-            startActivity(openWithIntent(file, userDrive))
+            startActivity(openWithIntent)
         } catch (e: ActivityNotFoundException) {
             showToast(R.string.errorNoSupportingAppFound)
         }
     }
 
     fun Context.openWithIntent(file: File, userDrive: UserDrive = UserDrive()): Intent {
         val (cloudUri, uri) = file.getCloudAndFileUris(this, userDrive)
-        return Intent().apply {
+        val flags = Intent.FLAG_GRANT_READ_URI_PERMISSION or
+                Intent.FLAG_GRANT_WRITE_URI_PERMISSION or
+                Intent.FLAG_GRANT_PERSISTABLE_URI_PERMISSION
+        return openWithIntent(uri, contentResolver.getType(cloudUri), flags)
+    }
+
+    fun Context.openWithIntent(uri: Uri, type: String?, flags: Int): Intent {
+        val openWithIntent = Intent().apply {
             action = Intent.ACTION_VIEW
-            flags = Intent.FLAG_GRANT_READ_URI_PERMISSION or
-                    Intent.FLAG_GRANT_WRITE_URI_PERMISSION or
-                    Intent.FLAG_GRANT_PERSISTABLE_URI_PERMISSION
-            setDataAndType(uri, contentResolver.getType(cloudUri))
+            this.flags = flags
+            setDataAndType(uri, type)
+        }
+
+        //We only do that when we try to openWith with a PDF because we have our own PDF reader
+        //Title in the chooser might not be displayed, at the discretion of the brand manufacturer
+        //So we keep the ACTION_VIEW for every type of files EXCEPT PDF files
```

Comment:
If we want to stay coherent we usually add spaces after a // and before the first letter of a line comment in the projects